test/python/circuit/test_unitary.py

```diff
@@ -267,3 +270,9 @@ def test_qasm_2q_unitary(self):
                         "custom_gate q0[0],q0[1];\n" \
                         "custom_gate q0[1],q0[0];\n"
         self.assertEqual(expected_qasm, qc.qasm())
+
+    def test_unitary_decomposition(self):
+        """Test decomposition for unitary gates over 2 qubits."""
+        qc = QuantumCircuit(3)
+        qc.unitary(random_unitary(8, seed=42), [0, 1, 2])
+        matrix_equal(Operator(qc).data, Operator(qc.decompose()).data)
```

Comment:
this isn't asserting anything

```suggestion
        self.assertTrue(Operator(qc).equiv(Operator(qc.decompose())))
```